pkgs/applications/audio/ChowCentaur/default.nix

```
@@ -0,0 +1,41 @@
+{ stdenv, lib, fetchFromGitHub, cmake, pkg-config, cairo, libxkbcommon, xcbutilcursor, xcbutilkeysyms, xcbutil, libXrandr, libXinerama, libXcursor, alsa-lib, libjack2 ,lv2, gcc-unwrapped
+}:
+
+stdenv.mkDerivation rec {
+  pname = "ChowCentaur";
+  version = "1.4.0";
+
+  src = fetchFromGitHub {
+    owner = "jatinchowdhury18";
+    repo = "KlonCentaur";
+    rev = "v${version}";
+    sha256 = "0mrzlf4a6f25xd7z9xanpyq7ybb4al01dzpjsgi0jkmlmadyhc4h";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ cmake pkg-config ];
+  buildInputs = [ cairo libxkbcommon xcbutilcursor xcbutilkeysyms xcbutil libXrandr libXinerama libXcursor alsa-lib libjack2 lv2 ];
+
+  cmakeFlags = [ "-DCMAKE_AR=${gcc-unwrapped}/bin/gcc-ar"
+                 "-DCMAKE_RANLIB=${gcc-unwrapped}/bin/gcc-ranlib"
+                 "-DCMAKE_NM=${gcc-unwrapped}/bin/gcc-nm" ];
+
```

```suggestion
  cmakeFlags = [
    "-DCMAKE_AR=${gcc-unwrapped}/bin/gcc-ar"
    "-DCMAKE_RANLIB=${gcc-unwrapped}/bin/gcc-ranlib"
    "-DCMAKE_NM=${gcc-unwrapped}/bin/gcc-nm"
  ];
```
you are sure about unwrapped?

---

Without these lines, I get [this output](https://sebsauvage.net/paste/?65018a6d52c5a3c2#atT0J83wGziqR588iAa2hLkVZr6HYu+0ExSMlagtLrs=), and when I replace them with
```nix
  cmakeFlags = [ "-DCMAKE_AR=${stdenv.cc.targetPrefix}ar"
                 "-DCMAKE_RANLIB=${stdenv.cc.targetPrefix}ranlib"
                 "-DCMAKE_NM=${stdenv.cc.targetPrefix}nm" ];
```
as suggested [here](https://github.com/NixOS/nixpkgs/pull/136762#discussion_r706410624), I get:
```shell
nix-env -f $NIXPKGS -iA ChowCentaur  --max-jobs 8 --builders ''

replacing old 'ChowCentaur-1.4.0'
installing 'ChowCentaur-1.4.0'
this derivation will be built:
  /nix/store/lsgdpl99v42y94dcm7i8crppg5pdldh6-ChowCentaur-1.4.0.drv
building '/nix/store/lsgdpl99v42y94dcm7i8crppg5pdldh6-ChowCentaur-1.4.0.drv'...
unpacking sources
unpacking source archive /nix/store/yl7aq9cp6w0zf539qibspkz3ka937ys2-source
source root is source
patching sources
configuring
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/zn03vgxh03j9xz689zya5szfwncdbf58-ChowCentaur-1.4.0/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/zn03vgxh03j9xz689zya5szfwncdbf58-ChowCentaur-1.4.0/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/zn03vgxh03j9xz689zya5szfwncdbf58-ChowCentaur-1.4.0/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/zn03vgxh03j9xz689zya5szfwncdbf58-ChowCentaur-1.4.0/share/doc/ChowCentaur -DCMAKE_INSTALL_INFODIR=/nix/store/zn03vgxh03j9xz689zya5szfwncdbf58-ChowCentaur-1.4.0/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/zn03vgxh03j9xz689zya5szfwncdbf58-ChowCentaur-1.4.0/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/zn03vgxh03j9xz689zya5szfwncdbf58-ChowCentaur-1.4.0/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/zn03vgxh03j9xz689zya5szfwncdbf58-ChowCentaur-1.4.0/include -DCMAKE_INSTALL_SBINDIR=/nix/store/zn03vgxh03j9xz689zya5szfwncdbf58-ChowCentaur-1.4.0/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/zn03vgxh03j9xz689zya5szfwncdbf58-ChowCentaur-1.4.0/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/zn03vgxh03j9xz689zya5szfwncdbf58-ChowCentaur-1.4.0/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/strip -DCMAKE_RANLIB=/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ranlib -DCMAKE_AR=/nix/store/v8imx1nvyz0hgvx9cbcmh6gp4ngw3ffj-binutils-2.35.1/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/zn03vgxh03j9xz689zya5szfwncdbf58-ChowCentaur-1.4.0 -DCMAKE_AR=ar -DCMAKE_RANLIB=ranlib -DCMAKE_NM=nm
-- The C compiler identification is GNU 10.3.0
-- The CXX compiler identification is GNU 10.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/4xs1xyj8728yvh9y5v6ji819kwgfy2fv-gcc-wrapper-10.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/4xs1xyj8728yvh9y5v6ji819kwgfy2fv-gcc-wrapper-10.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: pkg-config (found version "0.29.2")
-- Checking for module 'libcurl'
--   No package 'libcurl' found
-- Checking for modules 'webkit2gtk-4.0;gtk+-x11-3.0'
--   No package 'webkit2gtk-4.0' found
--   No package 'gtk+-x11-3.0' found
-- Checking for module 'alsa'
--   Found alsa, version 1.2.5.1
-- Checking for module 'freetype2'
--   Found freetype2, version 24.0.18
-- Configuring juceaide
-- Building juceaide
-- Exporting juceaide
-- VERSION for JUCE: 6.0.8
-- RTNeural - Using xsimd backend
-- Building LV2 plugin format
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
building
build flags: -j8 -l8 SHELL=/nix/store/dpjnjrqbgbm8a5wvi1hya01vd8wyvsq4-bash-4.4-p23/bin/bash
Scanning dependencies of target RTNeural
Scanning dependencies of target ChowCentaur_LV2_lv2_ttl_generator
[  1%] Generating juce_binarydata_BinaryData/JuceLibraryCode/BinaryData1.cpp, juce_binarydata_BinaryData/JuceLibraryCode/BinaryData2.cpp, juce_binarydata_BinaryData/JuceLibraryCode/BinaryData3.cpp, juce_binarydata_BinaryData/JuceLibraryCode/BinaryData4.cpp, juce_binarydata_BinaryData/JuceLibraryCode/BinaryData5.cpp, juce_binarydata_BinaryData/JuceLibraryCode/BinaryData6.cpp, juce_binarydata_BinaryData/JuceLibraryCode/BinaryData7.cpp, juce_binarydata_BinaryData/JuceLibraryCode/BinaryData.h
Scanning dependencies of target toms917
[  3%] Building CXX object ChowCentaur/CMakeFiles/ChowCentaur_LV2_lv2_ttl_generator.dir/__/modules/JUCE/extras/Build/lv2_ttl_generator/lv2_ttl_generator.c.o
[  3%] Building CXX object modules/toms917/CMakeFiles/toms917.dir/toms917.cpp.o
Scanning dependencies of target BinaryData
[  7%] Building CXX object ChowCentaur/CMakeFiles/BinaryData.dir/juce_binarydata_BinaryData/JuceLibraryCode/BinaryData3.cpp.o
[  7%] Building CXX object ChowCentaur/CMakeFiles/BinaryData.dir/juce_binarydata_BinaryData/JuceLibraryCode/BinaryData1.cpp.o
[  7%] Building CXX object ChowCentaur/CMakeFiles/BinaryData.dir/juce_binarydata_BinaryData/JuceLibraryCode/BinaryData4.cpp.o
[  8%] Building CXX object ChowCentaur/CMakeFiles/BinaryData.dir/juce_binarydata_BinaryData/JuceLibraryCode/BinaryData5.cpp.o
[ 10%] Building CXX object ChowCentaur/CMakeFiles/BinaryData.dir/juce_binarydata_BinaryData/JuceLibraryCode/BinaryData2.cpp.o
[ 11%] Building CXX object modules/RTNeural/RTNeural/CMakeFiles/RTNeural.dir/RTNeural.cpp.o
[ 12%] Building CXX object ChowCentaur/CMakeFiles/BinaryData.dir/juce_binarydata_BinaryData/JuceLibraryCode/BinaryData6.cpp.o
[ 13%] Building CXX object ChowCentaur/CMakeFiles/BinaryData.dir/juce_binarydata_BinaryData/JuceLibraryCode/BinaryData7.cpp.o
[ 15%] Linking CXX executable ChowCentaur_LV2_lv2_ttl_generator
[ 16%] Linking CXX static library libBinaryData.a
Error running link command: No such file or directory
make[2]: *** [ChowCentaur/CMakeFiles/BinaryData.dir/build.make:225: ChowCentaur/libBinaryData.a] Error 2
make[1]: *** [CMakeFiles/Makefile2:365: ChowCentaur/CMakeFiles/BinaryData.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 16%] Built target ChowCentaur_LV2_lv2_ttl_generator
[ 17%] Linking CXX static library libtoms917.a
Error running link command: No such file or directory
make[2]: *** [modules/toms917/CMakeFiles/toms917.dir/build.make:104: modules/toms917/libtoms917.a] Error 2
make[1]: *** [CMakeFiles/Makefile2:338: modules/toms917/CMakeFiles/toms917.dir/all] Error 2
[ 18%] Linking CXX static library libRTNeural.a
Error running link command: No such file or directory
make[2]: *** [modules/RTNeural/RTNeural/CMakeFiles/RTNeural.dir/build.make:104: modules/RTNeural/RTNeural/libRTNeural.a] Error 2
make[1]: *** [CMakeFiles/Makefile2:311: modules/RTNeural/RTNeural/CMakeFiles/RTNeural.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
error: builder for '/nix/store/lsgdpl99v42y94dcm7i8crppg5pdldh6-ChowCentaur-1.4.0.drv' failed with exit code 2;
       last 10 log lines:
       > [ 16%] Built target ChowCentaur_LV2_lv2_ttl_generator
       > [ 17%] Linking CXX static library libtoms917.a
       > Error running link command: No such file or directory
       > make[2]: *** [modules/toms917/CMakeFiles/toms917.dir/build.make:104: modules/toms917/libtoms917.a] Error 2
       > make[1]: *** [CMakeFiles/Makefile2:338: modules/toms917/CMakeFiles/toms917.dir/all] Error 2
       > [ 18%] Linking CXX static library libRTNeural.a
       > Error running link command: No such file or directory
       > make[2]: *** [modules/RTNeural/RTNeural/CMakeFiles/RTNeural.dir/build.make:104: modules/RTNeural/RTNeural/libRTNeural.a] Error 2
       > make[1]: *** [CMakeFiles/Makefile2:311: modules/RTNeural/RTNeural/CMakeFiles/RTNeural.dir/all] Error 2
       > make: *** [Makefile:149: all] Error 2
       For full logs, run 'nix log /nix/store/lsgdpl99v42y94dcm7i8crppg5pdldh6-ChowCentaur-1.4.0.drv'.
```